servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectConnectionFactoryFilter.java

```diff
@@ -74,14 +70,21 @@ private ProxyFilter(final ConnectionFactory<ResolvedAddress, C> delegate) {
 
         @Override
         public Single<C> newConnection(final ResolvedAddress resolvedAddress) {
-            return delegate().newConnection(resolvedAddress).flatMap(c ->
+            return delegate().newConnection(resolvedAddress).flatMap(c -> {
+            try {
                 // We currently only have access to a StreamingHttpRequester, which means we are forced to provide an
                 // HttpExecutionStrategy. Because we can't be sure if there is any blocking code in the connection
                 // filters we use the default strategy which should offload everything to be safe.
-                c.request(defaultStrategy(),
-                            reqRespFactory.connect(connectAddress).addHeader(CONTENT_LENGTH, ZERO))
+                return c.request(defaultStrategy(), c.connect(connectAddress).addHeader(CONTENT_LENGTH, ZERO))
                  .flatMap(response -> {
-                if (SUCCESSFUL_2XX.contains(response.status())) {
+                    // Drain response payload body asynchronously as we are not interested in it:
+                    response.payloadBodyAndTrailers().ignoreElements().subscribe();
```

Comment:
Dangling subscribes are usually a bad idea. As this code is not composing the subscribe, if the returned `Single` from `newConnection` is cancelled, we will still keep consuming the response. Worse if the response does not complete, we will keep this subscribe around forever (and so the resources associated with the subscribe). 

What we were doing before, i.e. composing the response draining with the `Single` returned from this `flatMap` is the correct thing to do.